source/common/grpc/async_client_impl.cc

```diff
@@ -321,6 +337,23 @@ void AsyncStreamImpl::cleanup() {
     dispatcher_->deferredDelete(
         LinkedObject<AsyncStreamImpl>::removeFromList(parent_.active_streams_));
   }
+  remote_close_timer_ = nullptr;
+}
+
+void AsyncStreamImpl::waitForRemoteCloseAndDelete() {
+  if (!waiting_to_delete_on_remote_close_) {
+    waiting_to_delete_on_remote_close_ = true;
+
+    if (options_.sidestream_watermark_callbacks != nullptr) {
+      stream_->removeWatermarkCallbacks();
+      options_.sidestream_watermark_callbacks = nullptr;
+    }
+    remote_close_timer_ = dispatcher_->createTimer([this] {
+      waiting_to_delete_on_remote_close_ = false;
```

Comment:
naive Q: what does this line do?

Reply:
I think you just answered it: https://github.com/envoyproxy/envoy/pull/38661/files#r2002338431